dags/cluster_dag.py

```diff
@@ -56,14 +56,18 @@ def cluster_control():
         project_id = gapi.get_project_id()
         cluster_info = json.loads(BaseHook.get_connection("InstanceGroups").extra)
         target_sizes = Variable.get("cluster_target_size", deserialize_json=True)
+        min_sizes = Variable.get("cluster_min_size", deserialize_json=True)
     except:
         slack_message(":exclamation:Failed to load the cluster information from connection {}".format("InstanceGroups"), notification=True)
         return
     for key in cluster_info:
         if key in target_sizes:
             print(f"processing cluster: {key}")
-            if target_sizes[key] != 0:
 
+            min_size = gapi.get_min_size(key, min_sizes)[0]
```

Comment:
Putting the `get_min_size` function inside gapi and calling it from outside is confusing, the function just fetch the variable and do not really use any google specific function

Reply:
Yeah. I thought about this. I just wasn't sure where else to put this or whether you'd approve of another module for this function. One thought I just had - what if I move `airflow_api.py` to `common` and put there instead?

Reply:
I think it is easiest to create another module, I do not think all of the functions in `airflow_api.py` should be moved to `common`, some of them like triggering dags, cancel dags are really hacks using unpublished api. Feel free to create a new module just for the "warm up" functionality.
I also wonder since you already limit the number of instance to `min_size` in `ramp_down_cluster`, is that already enough for warming up? More broadly is it possible to hide the `min_size` detail from the cluster dag.

Reply:
I think removing it from the cluster_management dag would work if we accept two consequences:
1. The user would get a lot of misleading slack messages when the `target_size` doesn't match how the cluster actually behaves
2. We would need a way for the warm up function to ramp the cluster up initially, probably a separate DAG with one node for `scale_up_cluster_op`.

Reply:
OK, then just add a new module for the `min_size` call and the timestamp related code.